src/server/services/toeristische-verhuur/toeristische-verhuur.ts

```diff
@@ -0,0 +1,299 @@
+import { subMonths } from 'date-fns';
+import memoize from 'memoizee';
+import { generatePath } from 'react-router-dom';
+import { FeatureToggle, Themas } from '../../../universal/config';
+import { AppRoutes } from '../../../universal/config/routes';
+import {
+  dateFormat,
+  dateSort,
+  isDateInPast,
+  isRecentNotification,
+} from '../../../universal/helpers';
+import {
+  apiDependencyError,
+  apiSuccessResult,
+  getFailedDependencies,
+  getSettledResult,
+} from '../../../universal/helpers/api';
+import {
+  NOTIFICATION_REMINDER_FROM_MONTHS_NEAR_END,
+  isNearEndDate2,
+} from '../../../universal/helpers/vergunningen';
+import { MyNotification } from '../../../universal/types';
+import { DEFAULT_API_CACHE_TTL_MS } from '../../config';
+import { AuthProfileAndToken } from '../../helpers/app';
+import { BBVergunning, fetchBBVergunning } from './bb-vergunning';
+import {
+  ToeristischeVerhuurRegistratieDetail,
+  fetchRegistraties,
+} from './lvv-registratie';
+import {
+  VakantieverhuurVergunning,
+  fetchVakantieverhuurVergunningen,
+} from './vakantieverhuur-vergunning';
+
+export function hasOtherActualVergunningOfSameType(
+  items: Array<VakantieverhuurVergunning | BBVergunning>,
+  item: VakantieverhuurVergunning | BBVergunning,
+  dateNow: Date = new Date()
+): boolean {
+  return items.some(
+    (otherVergunning: VakantieverhuurVergunning | BBVergunning) =>
+      otherVergunning.titel === item.titel &&
+      otherVergunning.zaaknummer !== item.zaaknummer &&
+      !isDateInPast(otherVergunning.datumTot, dateNow)
+  );
+}
+
+async function fetchAndTransformToeristischeVerhuur(
+  requestID: requestID,
+  authProfileAndToken: AuthProfileAndToken,
+  profileType: ProfileType = 'private'
+) {
+  if (!FeatureToggle.toeristischeVerhuurActive) {
+    return apiSuccessResult({
+      vakantieverhuurVergunningen: [],
+      bbVergunningen: [],
+      lvvRegistraties: [],
+    });
+  }
+  const lvvRegistratiesRequest =
+    profileType === 'commercial'
+      ? Promise.resolve(apiSuccessResult([]))
+      : fetchRegistraties(requestID, authProfileAndToken);
+
+  const bbVergunningenRequest = fetchBBVergunning(
+    requestID,
+    authProfileAndToken
+  );
+
+  const vakantieverhuurVergunningenRequest = fetchVakantieverhuurVergunningen(
+    requestID,
+    authProfileAndToken
+  );
+
+  const [
+    lvvRegistratiesResponse,
+    vakantieverhuurVergunningenResponse,
+    bbVergunningenResponse,
+  ] = await Promise.allSettled([
+    lvvRegistratiesRequest,
+    vakantieverhuurVergunningenRequest,
+    bbVergunningenRequest,
+  ]);
+
+  const lvvRegistraties = getSettledResult(lvvRegistratiesResponse);
```

Comment:
getSettledResult is heeft een redundant variable initializer;  let errorMessage = result.reason.toString();